### PR TITLE
Copter: stop sending POSITION_TARGET_LOCAL_NED in guided-angle mode 

### PR DESCRIPTION
**Pull request by peterbarker**
The message is not relevant when doing angle-control.

This issue became apparent when moving Guided_Mode to be an enum-class on mode-guided, something I wanted to do as part of having a "landing" submode in guided (rather than it currently just switching to mode-land.  In any case, using `case` statements means people need to think about things when adding a new submode.

### CODE REVIEW

**Review by khancyr — COMMENTED** [review]
LGTM, nice cleanup

**Review by khancyr — APPROVED** [review]

File: ArduCopter/mode.h [hunk 1]
```diff
@@ -846,7 +846,15 @@ class ModeGuided : public Mode {
 
     bool do_user_takeoff_start(float takeoff_alt_cm) override;
 
-    GuidedMode mode() const { return guided_mode; }
+    enum class SubMode {
+        TakeOff,
```
Comment (khancyr) [inline ArduCopter/mode.h:850]: all capitals like other enum ?

Comment (peterbarker) [reply ArduCopter/mode.h:850]: We've previously decided we'll accept either, and I vaguely recall @rmackay9  resisted a similar request for change to all-caps.

I prefer the all-caps thing, mind you.

**Review by tridge — APPROVED** [review]

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**